compiler/visq/visq

```diff
@@ -0,0 +1,206 @@
+    cmd += ['--input_model', str(model)]
+    cmd += ['--analysis', str(analysis)]
+    if data != None:
+        cmd += ['--input_data', str(data)]
+    cmd += ['--analysis_args', str(save_dir)]
+
+    try:
+        subprocess.run(cmd, capture_output=True, check=True)
+    except subprocess.CalledProcessError as e:
+        print('Error raised while running the below command')
+        print(' '.join(cmd))
+        print(e.output)
+        raise
+
+
+def _fake_quantize(input_model, output_model):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    circle_quantizer_path = os.path.join(dir_path, 'circle-quantizer')
+    cmd = [circle_quantizer_path]
+    cmd += ['--fake_quantize']
+    cmd += [str(input_model)]
+    cmd += [str(output_model)]
+
+    try:
+        subprocess.run(cmd, capture_output=True, check=True)
+    except subprocess.CalledProcessError as e:
+        print('Error raised while running the below command')
+        print(' '.join(cmd))
+        print(e.output)
+        raise
+
+
+# Recursively visit items and check if there is Infinity or NaN
+def _check_float(item):
+    if isinstance(item, dict):
+        for v in item.values():
+            _check_float(v)
+    if isinstance(item, list):
+        for v in item:
+            _check_float(v)
+    if isinstance(item, float):
+        if item == -float('inf') or item == float('inf'):
+            raise RuntimeError('Infinite value detected. Value must be float')
+        if math.isnan(item):
+            raise RuntimeError('NaN value detected. Value must be float')
+
+
+def _build_json(**kwargs):
+    # model: string
+    # metric: string
+    # colorscheme: list ['b': begin, 'e': end, 'c':color]
+    # error: dict {tensor_name:error}
+    assert 'metric' in kwargs
+    assert 'model' in kwargs
+    assert 'colorscheme' in kwargs
+    assert 'error' in kwargs
+
+    meta = {}
+    meta["model"] = Path(kwargs['model']).name
+    meta["metric"] = kwargs['metric']
+    meta["colorscheme"] = pretty_float(kwargs['colorscheme'])
+    result = {}
+    result["meta"] = meta
+    # Why list? To support multiple subgraphs
+    result["error"] = [pretty_float(kwargs['error'])]
+
+    # Invariants
+    _check_float(meta["colorscheme"])
+    _check_float(result["error"])
+    return result
+
+
+def main():
+    # parse arguments
+    parser = _get_parser()
+    args = parser.parse_args()
+    _verify_args(args)
+
+    fp32_model = args.fp32_circle
+    q_model = args.q_circle
+    data = None
+    if valid_attr(args, 'data'):
+        data = args.data
+
+    curr_dir = Path(__file__).parent.resolve()
+    dump_fp32_py = curr_dir / 'visqlib' / 'DumpFP32FM.py'
+    dump_fq_py = curr_dir / 'visqlib' / 'DumpFakeQuantFM.py'
+
+    with tempfile.TemporaryDirectory() as model_dir, \
+         tempfile.TemporaryDirectory() as fp32_dir, \
+         tempfile.TemporaryDirectory() as fq_dir:
+        fq_model = model_dir + '/fq_model.circle'
+
+        # Step 1. Fake quantize q circle
```

Comment:
Q) `q` here is `quantized` ?
And about `circle` here, other comments use `model`, so IIUC, how about `Fake quantize quantized circle model` ?